core/src/actions_normalize.cpp

```diff
@@ -94,75 +65,61 @@ bool km::core::actions_normalize(
     cached_context.
   */
 
-  /*
-    Initialization
-  */
-
-  UErrorCode icu_status = U_ZERO_ERROR;
-  const icu::Normalizer2 *nfc = icu::Normalizer2::getNFCInstance(icu_status);
-  assert(U_SUCCESS(icu_status));
-  if(!U_SUCCESS(icu_status)) {
-    DebugLog("getNFCInstance failed with %x", icu_status);
-    return false;
-  }
-
-  const icu::Normalizer2 *nfd = icu::Normalizer2::getNFDInstance(icu_status);
-  assert(U_SUCCESS(icu_status));
-  if(!U_SUCCESS(icu_status)) {
-    DebugLog("getNFDInstance failed with %x", icu_status);
+  std::u32string output(actions.output);
+  std::u32string cached_context_string, app_context_string;
+  if (!context_items_to_unicode_string(cached_context, cached_context_string)) {
     return false;
   }
-
-  icu::UnicodeString output = icu::UnicodeString::fromUTF32(reinterpret_cast<const UChar32*>(actions.output), -1);
-  icu::UnicodeString cached_context_string = context_items_to_unicode_string(cached_context);
-  icu::UnicodeString app_context_string = context_items_to_unicode_string(app_context);
-  assert(!output.isBogus());
-  assert(!cached_context_string.isBogus());
-  assert(!app_context_string.isBogus());
-  if(output.isBogus() || cached_context_string.isBogus() || app_context_string.isBogus()) {
+  if (!context_items_to_unicode_string(app_context, app_context_string)) {
```

Comment:
The new format of the method - returning an error code/flag instead of the ICU-string object directly - allows us to reuse its internal assertions + error-throwing instead of needing the previously-WET style being cleaned up here.

Assuming I'm parsing things correctly.  (Took a while to ferret this detail out.)